ariadne/wsgi.py

```diff
@@ -149,22 +154,35 @@ def extract_data_from_multipart_request(self, environ: dict) -> Any:
         return combine_multipart_data(operations, files_map, form)
 
     def execute_query(self, environ: dict, data: dict) -> GraphQLResult:
+        context_value = self.get_context_for_request(environ)
+        middleware = self.get_middleware_for_request(environ, context_value)
+
         return graphql_sync(
             self.schema,
             data,
-            context_value=self.get_context_for_request(environ),
+            context_value=context_value,
             root_value=self.root_value,
             debug=self.debug,
             logger=self.logger,
             error_formatter=self.error_formatter,
-            middleware=self.middleware,
+            middleware=middleware,
         )
 
-    def get_context_for_request(self, environ: dict) -> Any:
+    def get_context_for_request(self, environ: dict) -> Optional[ContextValue]:
         if callable(self.context_value):
             return self.context_value(environ)
         return self.context_value or environ
 
+    def get_middleware_for_request(
+        self, environ: dict, context: Optional[ContextValue]
+    ) -> Optional[MiddlewareManager]:
+        middleware = self.middleware
+        if callable(middleware):
+            middleware = middleware(environ, context)
+        if middleware:
+            return MiddlewareManager(*middleware)
+        return None
```

Comment:
Not really necessary.

Reply:
It's either that or `pylint: disable=...`, so I'll rather `return None` as former will remove useful safety check from pylint.